pkg/virt-controller/services/template.go

```diff
@@ -539,6 +540,29 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, i
 			TopologySpreadConstraints:     vmi.Spec.TopologySpreadConstraints,
 		},
 	}
+	tscFrequency := nsr.TSCFrequency()
+	if tscFrequency != 0 {
+		pod.Spec.Affinity = &k8sv1.Affinity{
```

Comment:
just a premature suggestion:
`pod.Spec.Affinity` is overridden on line 556 in case vmi has a specific affinity

Reply:
Not premature at all, great catch, thank you!! :)